simexpal/base.py

```diff
@@ -419,6 +423,46 @@ def export_successful_experiments(self):
 	def export_failed_experiments(self):
 		return self.export_experiments([s for s in Status if s.is_negative])
 
+	def query_slurm(self):
+		if not self.slurm_queried:
+			# -h omits the header line.
+			# -r outputs one job array element per line.
+			output = subprocess.check_output(['squeue', '-h', '-r'])
+			output = output.decode().splitlines()
+
+			if len(output) > 0:
+				output = [entry.split() for entry in output]
+				for entry in output:
+					entry_jobid = entry[0]
+					entry_state = entry[4]
+					if entry_state == 'PD':
+						status = Status.SUBMITTED
+					elif entry_state in ['R', 'CG']:
+						status = Status.STARTED
+					else:
+						status = Status.BROKEN
+
+					self.slurm_queried_jobs[entry_jobid] = status
+
+			if len(self.slurm_queried_jobs) > 0:
+				self.slurm_queried = True
+
+	def pop_slurm_queried_jobs(self, jobid):
+		# Returns 'failed' for jobs not listed in self.slurm_queried_jobs.
+		# This only holds for jobs that would receive the status
+		# 'submitted' or 'started' in Run._update_status_cache_dict().
+		status = self.slurm_queried_jobs.pop(jobid, Status.FAILED)
+		if len(self.slurm_queried_jobs) == 0:
+			self.slurm_queried = False
+
+		return status
```

Comment:
This function is a bit strange. Why do we pop the status instead of just returning it?

Reply:
Hm, I think I wanted to track whether the status of a job was already queried or not, but in hindsight it is probably not necessary

Reply:
done